modules/manifold/src/components/ui/segment-filters-control/segment-filter.js

```diff
@@ -0,0 +1,160 @@
+    filter: FILTER.isRequired,
+    columnDef: FIELD.isRequired,
+    onUpdateFilterValue: PropTypes.func,
+  };
+
+  static defaultProps = {
+    onUpdateFilterValue: () => {},
+  };
+
+  state = {
+    selectedDomain: [],
+  };
+
+  _computeOptions = memoize(
+    columnDef => {
+      return columnDef.domain.map(domainVal => ({
+        id: domainVal,
+      }));
+    },
+    columnDef => columnDef.tableFieldIndex
+  );
+
+  _getValuesFromOptions = options => options.map(o => o.id);
+
+  _onUpdateFilterValue = ({value, type}) => {
+    const {segmentId, filterId, onUpdateFilterValue} = this.props;
+    const _value =
+      type === 'select' ? this._getValuesFromOptions(value) : value;
+    onUpdateFilterValue(segmentId, filterId, _value);
+  };
+
+  _renderFilterControl = () => {
+    const {filter, columnDef} = this.props;
+    if (!filter || !columnDef) {
+      return null;
+    }
+    const min = columnDef.domain[0];
+    const max = columnDef.domain[columnDef.domain.length - 1];
+
+    switch (filter.type) {
+      case FILTER_TYPE.RANGE:
+        return (
+          <Slider
+            key={filter.key}
+            min={min}
+            max={max}
+            value={filter.value}
+            onChange={this._onUpdateFilterValue}
+            overrides={{
+              Tick: {
+                component: TickOverrides,
+                style: {
+                  ...THEME.typography.font100,
+                  paddingLeft: THEME.sizing.scale300,
+                  paddingRight: THEME.sizing.scale300,
+                },
+              },
+              ThumbValue: {style: {...THEME.typography.font100}},
+              Track: {
+                style: {
+                  paddingTop: THEME.sizing.scale900,
+                  paddingBottom: THEME.sizing.scale500,
+                  paddingLeft: THEME.sizing.scale300,
+                  paddingRight: THEME.sizing.scale300,
+                },
+              },
+            }}
+          />
+        );
+      case FILTER_TYPE.INCLUDE:
+        return (
+          <Select
+            key={filter.key}
+            size={SIZE.compact}
+            options={this._computeOptions(columnDef)}
+            value={filter.value.map(val => ({id: val}))}
+            labelKey="id"
+            valueKey="id"
+            searchable={true}
+            multi={true}
+            onChange={this._onUpdateFilterValue}
+            overrides={{
+              Root: {style: {paddingBottom: '10px'}},
+              MultiValue: {
+                props: {variant: VARIANT.outlined},
+                style: {borderWidth: '1px'},
+              },
+            }}
+          />
+        );
+      default:
+        return <div>{`Feature type ${columnDef.type} not supported.`}</div>;
+    }
+  };
+
+  render() {
+    const {filter} = this.props;
+    if (!filter) {
+      return null;
+    }
+
+    return (
+      <Container>
+        <Header>{filter.name}</Header>
+        <ThemeProvider theme={localTheme}>
```

Comment:
Is this theme/style overriding meant to meet the current design?
If yes, we should ask Chun to align it with the lastest BaseWeb norms if we were to go with BaseUI.
Otherwise, we will have to maintain multiple sets of styles.

Reply:
This style could go to our main style in theme.js. I've only overridden it here because the filters look exceptionally unacceptable if using default baseui styles (compared to other parts of the UI). Note that style-overriding is a common pattern enabled by baseui https://baseweb.design/guides/theming/. Will run it through designer to see whether we put this overriding as default or only in part of the UI.